check static "host" parameters contain valid hostname

kerolasa opened the pull request:
Reported-by: Pawel Kowalik <kowalik@denic.de>
Fixes: https://github.com/Domain-Connect/dc-template-linter/issues/13

thoag-godaddy commented:
I think this is a result of the discussion in https://github.com/Domain-Connect/Templates/pull/568
In that PR it was the rfc import of "sub domain" that made the underscores invalid for the dkim selector.

In my research there is no indication that a DNS name containing underscores beyond position 1 is invalid. It is true that host names cannot have said underscores, but not necessarily true for a domain name (rfc1034).

pawel-kow commented on internal/dctl.go:
```diff
@@ -105,6 +106,7 @@ var dctlToString = map[DCTL]string{
 	DCTL1022: "invalid hostname",
 	DCTL1023: "duplicate record entry",
 	DCTL1024: "use of %host% variable name is problematic",
+	DCTL1025: "use of underscore in detected in a host: element",
```
One in too much

kerolasa commented:
@thoag-godaddy I agree. Underscore is merely something one should review more carefully. I updated the change to lower the messaging level from Warning to Info, and added a bit of text to commit message telling it's not end of the world if underscore is detected -- one should just think about it before moving forward.
